documentation/src/docs/security/securing-distributed-mode-cluster.md

```diff
@@ -0,0 +1,311 @@
+4. Preparing a keystore containing the signed server certificate and the CA's certificate
+5. Exporting the server certificate's private key
+
+Note:
+
+* For services running on the same host, you may use the same certificate if those services are exposed using the same hostname/IP address. You may even use wildcard certificates to share certificates across hosts. However, it is strongly recommended that you use seperate certificates for each service.
+
+Let's see the steps in detail below.
+
+1. Generate keys and certificates for each service. This certificate is used for establishing TLS as well as for verifying the server's identity.
+
+   ```bash
+   $ keytool -keystore controller01.jks\
+    -genkey -keyalg RSA -keysize 2048 -keypass changeit\
+    -alias controller01 -validity 365\
+    -dname "CN=controller01.pravega.io, OU=Pravega, O=Company, L=Locality, S=StateOrProvince, C=Country"\
+    -ext san=dns:controller01.pravega.io,ip:<IP_ADDRESS>\
+    -storepass changeit
+
+   # Optionally, verify the contents of the generated file:
+   $ keytool -list -v -keystore controller01.jks -storepass changeit
+   ```
+
+2. Generate CSR for each service. It helps to think of a CSR as an application for getting a certificate signed by a trusted authority.
+
+   A CSR is typically generated on the same server on which the service is planned to be installed. Some organizations generate these outside of the servers in a central location and then distribute the resulting certificates to the services for use.
+
+   ```
+   $ keytool -keystore controller01.jks -alias controller01 -certreq -file controller01.csr \
+       -storepass changeit
+
+   # Optionally, inspect the contents of the CSR file. The CSR is created in Base-64 encoded PEM format.
+   $ openssl req -in controller01.csr -noout -text
+   ```
+
+3. Submit the CSR to a CA and obtain a signed certificate for each service.
+
+   If you are using a public or internal CA service, follow that CA's process for submitting the CSR and obtaining a signed certificate. To use the custom CA generated using the steps mentioned earlier or an internal CA certificate/key bundle, use the following command to generate a CA-signed server certificate in PEM format:
+
+   ```
+   $ openssl x509 -req -CA ca-cert -CAkey ca-key -in controller01.csr -out controller01.pem \
+        -days 3650 -CAcreateserial -passin pass:changeit
+   ```
+
+4. Prepare a keystore containing the signed server certificate and the CA's certificate.
+
+   ```
+   # Import the CA certificate into a new keystore file.
+   $ keytool -keystore controller01.server.jks -alias CARoot -noprompt \
+           -import -file ca-cert -storepass changeit
+
+   # Import the signed server certificate into the keystore.
+   $ keytool -keystore controller01.server.jks -alias controller01 -noprompt \
+           -import -file controller01.pem -storepass changeit
+   ```
+
+5. Export each server's key into a separate .pem file.
```

Comment:
`.pem`

Reply:
Changed it to PEM to keep it consistent.